src/configuration-client/src/main/java/ee/ria/xroad/common/conf/globalconf/ConfigurationDownloader.java

```diff
@@ -140,7 +142,7 @@ private List<ConfigurationLocation> getLocations(ConfigurationSource source) {
 
         preferLastSuccessLocation(source, result);
 
-        List<ConfigurationLocation> randomized = new ArrayList<>(source.getLocations());
+        List<ConfigurationLocation> randomized = new ArrayList<>(getLocationsPreferHttps(source));
         Collections.shuffle(randomized);
```

Comment:
With `shuffle`, the order with HTTPS links in front is lost. `getLocationsPreferHttps` should randomize lists before adding them together and `shuffle` should be removed from here.